src/foundry/foundry.js/hooks.d.ts

```diff
@@ -1,830 +1,804 @@
-    ) => boolean;
+    type GetSidebarDirectoryFolderContext = (jq: JQuery, folderOptions: ContextMenu.Item[]) => boolean | void;
 
     /**
-     * This is called after the {@link Game} pause is toggled
-     * @param paused - the new paused value of the Game
+     * A hook event that fires when the user modifies a global volume slider.
+     * The hook name needs to be customized to include the type of global volume being changed, one of:
+     * `globalPlaylistVolumeChanged`, `globalAmbientVolumeChanged`, or `globalInterfaceVolumeChanged`.
+     * @param volume - The new volume level
+     * @remarks The name for this hook is dynamically created by joining the name of the volume with 'Changed'.
      * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link Game#togglePause}
+     * @see {@link AudioHelper#_onChangeGlobalVolume}
      */
-    pauseGame: (paused: boolean) => unknown;
+    type GlobalVolumeChanged = (volumne: number) => unknown;
 
     /**
-     * This is called after the {@link Game} is fully set up.
+     * @remarks This is called when the user mouse is entering or leaving a hover state over a {@link PlaceableObject}.
+     * @param object - the PlaceableObject
+     * @param hover  - whether the mouse is hovering over the PlaceableObject
+     * @typeParam P  - the type of the PlaceableObject
+     * @remarks The name for this hook is dynamically created by joining 'hover' and the type name of the PlaceableObject.
      * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link Game#setupGame}
+     * @see {@link PlaceableObject#_onHoverIn}
+     * @see {@link PlaceableObject#_onHoverOut}
      */
-    ready: () => unknown;
+    type HoverPlaceableObject<P extends PlaceableObject = PlaceableObject> = (object: P, hover: boolean) => unknown;
 
     /**
-     * This is called as last step when rendering a {@link ChatMessage}.
-     * @param message     - the ChatMessage
-     * @param jq          - the JQuery of the rendered ChatMessage
-     * @param messageData - the data of the message
-     * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link ChatMessage#render}
+     * @remarks This is called after copying {@link PlaceableObject}s in a copy-paste action, but before embedding them
+     * into the {@link PlaceablesLayer}.
+     * @param copied - the originally copied PlaceableObjects
+     * @param pasted - the pasted copies with new coordinates
+     * @param P      - the type of the PlaceableObject
+     * @remarks The name for this hook is dynamically created by joining 'paste' with the type name of the
+     * PlaceableObject.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link PlaceablesLayer#pasteObjects}
      */
-    renderChatMessage: (message: ChatMessage, jq: JQuery, messageData: ChatMessage.MessageData) => unknown;
+    type PastePlaceableObject<P extends PlaceableObject = PlaceableObject> = (
+      copied: P[],
+      pasted: P[]
```

Comment:
This is actually an array of the source data of the documents that need to be created, I would also rather call it `toCreate` or something similar...